electrum/plugins/jade/jade.py

```diff
@@ -0,0 +1,464 @@
+                        # embedded in the multisig wallet registration record
+                        # NOTE: all cosigners have same path suffix
+                        path_suffix = wallet.get_address_index(txout.address)
+                        paths = [path_suffix] * wallet.n
+                        change[index] = {'multisig_name': multisig_name, 'paths': paths}
+                    else:
+                        # Pass entire path
+                        pubkey, path = self.find_my_pubkey_in_txinout(txout)
+                        change[index] = {'path':path, 'variant': txout.script_type}
+
+            # The txn itself
+            txn_bytes = bytes.fromhex(tx.serialize_to_network())
+
+            # Request Jade generate the signatures for our inputs.
+            # Change details are passed to be validated on the hw (user does not confirm)
+            self.handler.show_message(_("Please confirm the transaction details on your Jade device..."))
+            client = self.get_client()
+            signatures = client.sign_tx(txn_bytes, jade_inputs, change)
+            assert len(signatures) == len(tx.inputs())
+
+            # Inject signatures into tx
+            for index, (txin, signature) in enumerate(zip(tx.inputs(), signatures)):
+                pubkey, path = self.find_my_pubkey_in_txinout(txin)
+                if pubkey is not None and signature is not None:
+                    tx.add_signature_to_txin(txin_idx=index,
+                                            signing_pubkey=pubkey.hex(),
+                                            sig=signature.hex())
+        finally:
+            self.handler.finished()
+
+    @runs_in_hwd_thread
+    def show_address(self, sequence, txin_type):
+        self.handler.show_message(_("Showing address ..."))
+        try:
+            client = self.get_client()
+            bip32_path_prefix = self.get_derivation_prefix()
+            return client.show_address(bip32_path_prefix, sequence, txin_type)
+        finally:
+            self.handler.finished()
+
+    @runs_in_hwd_thread
+    def register_multisig(self, name, txin_type, sorted, threshold, signers):
+        self.handler.show_message(_("Please confirm the multisig wallet details on your Jade device..."))
+        try:
+            client = self.get_client()
+            return client.register_multisig(name, txin_type, sorted, threshold, signers)
+        finally:
+            self.handler.finished()
+
+    @runs_in_hwd_thread
+    def show_address_multi(self, multisig_name, paths):
+        self.handler.show_message(_("Showing address ..."))
+        try:
+            client = self.get_client()
+            return client.show_address_multi(multisig_name, paths)
+        finally:
+            self.handler.finished()
+
+
+class JadePlugin(HW_PluginBase):
+    keystore_class = Jade_KeyStore
+    minimum_library = (0, 0, 1)
+    client = None
```

Comment:
this line is copied dead code, see https://github.com/spesmilo/electrum/commit/aab8e664ed88a1f9c63e9a3d409e63775020626d

Reply:
Agreed, removed.